team-validator.js

```diff
@@ -819,6 +820,9 @@ class Validator {
 							// must be able to breed with father
 							if (!dexEntry.eggGroups.some(eggGroup => eggGroupsSet.has(eggGroup))) continue;
 
+							const moveGen = (dexEntry.learnset[moveid] ? dexEntry.learnset[moveid][0].charAt(0) : dexEntry.learnset['sketch'][0].charAt(0));
+							if (noPastGenBreeding && (dexEntry.tier.startsWith('Bank') || moveGen !== '7' || template.id === dexEntry.prevo || dexEntry.evos.indexOf(template.id) >= 0)) continue;
```

Comment:
Also, `template.id === dexEntry.prevo || dexEntry.evos.includes(template.id)` should be unnecessary:

```
// unless it's supposed to be self-breedable, can't inherit from self, prevos, etc
// only basic pokemon have egg moves, so by now all evolutions should be in alreadyChecked
if (!fromSelf && alreadyChecked[dexEntry.speciesid]) continue;
```

This code is supposed to handle that case.